package-lock.json.unit.test.mjs

```diff
@@ -28,4 +28,62 @@ describe('package-lock.json', () => {
       }
     })
   })
+
+  describe("Shared helper 'peerDependencies'", () => {
+    it('should maintain `puppeteer` bundled `@puppeteer/browsers` compatibility', () => {
+      const { peerDependencies } = packages['shared/helpers']
+      const { dependencies } = packages['node_modules/puppeteer']
+
+      // Bundled package `@puppeteer/browsers` versus peer dependencies
+      const installed = dependencies['@puppeteer/browsers']
+      const expected = peerDependencies['@puppeteer/browsers']
+
+      expect({
+        '@puppeteer/browsers': {
+          parent: 'shared/helpers',
+          installed,
+          expected,
+          compatible: semver.intersects(expected, installed)
+        }
+      }).toEqual({
+        '@puppeteer/browsers': {
+          parent: 'shared/helpers',
+          installed,
+          expected,
+          compatible: true
+        }
+      })
+    })
+
+    it("should maintain `puppeteer` workspace 'devDependencies' compatibility", () => {
+      const { peerDependencies } = packages['shared/helpers']
+
+      for (const name of /** @type {const} */ ([
+        'packages/govuk-frontend-review',
+        'packages/govuk-frontend',
+        'shared/helpers',
+        'shared/tasks'
+      ])) {
+        // Development package `puppeteer` versus peer dependencies
+        const installed = packages[name].devDependencies.puppeteer
+        const expected = peerDependencies['puppeteer-core']
```

Comment:
What's the reason for a different approach with this dependency? I'd have expected both to run the same check (ie. either both compare `puppeteer`'s dependencies with our versions, or both loop through our packages), so there's probably something I didn't consider that's worth mentionning so we know why the two tests are different in the future 😊

Reply:
Yeah course

We use `puppeteer` in lots of places so we loop through our packages:

```console
├─┬ @govuk-frontend/helpers@ -> ./shared/helpers
│ ├─┬ @axe-core/puppeteer@4.8.5
│ │ └── puppeteer@22.3.0 deduped
│ └── puppeteer@22.3.0 deduped
├─┬ @govuk-frontend/review@ -> ./packages/govuk-frontend-review
│ └── puppeteer@22.3.0 deduped
├─┬ @govuk-frontend/tasks@ -> ./shared/tasks
│ └── puppeteer@22.3.0 deduped
├─┬ govuk-frontend@5.2.0 -> ./packages/govuk-frontend
│ └── puppeteer@22.3.0 deduped
└── puppeteer@22.3.0
```

But for `@puppeteer/browsers` we only use it in our [browser `download.mjs`](https://github.com/alphagov/govuk-frontend/blob/035c12acd2d8ca880926a978d6991df92c659793/shared/helpers/jest/browser/download.mjs#L4-L5) script

```console
└─┬ @govuk-frontend/helpers@ -> ./shared/helpers
  └── @puppeteer/browsers@2.1.0
```

Reply:
**Note:** We only actually install `@puppeteer/browsers` directly to avoid the following ESLint error:

>"@puppeteer/browsers" is extraneous.eslint [n/no-extraneous-import](https://github.com/eslint-community/eslint-plugin-n/blob/HEAD/docs/rules/no-extraneous-import.md)

Rather than install directly, ESLint can locate `@puppeteer/browsers` using [`allowModules` paths](https://github.com/eslint-community/eslint-plugin-n/blob/HEAD/docs/rules/no-extraneous-import.md#allowmodules) but this risks missing `node_modules` in production because ESLint runs with all development packages present